app/Http/Controllers/User/UserControler.php

```diff
@@ -0,0 +1,109 @@
+<?php
+namespace App\Http\Controllers\User;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\CreateUserRequest;
+use Exception;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\View\View;
+use App\Models\User;
+
+class UserController extends Controller
+{
+    private $user;
+
+    // inject UserRepository as 
+    public function __construct(UserRepository $user)
+    {
+        $this->userRepository = $user;
+        $this->middleware('isAdmin')->only('create', 'store', 'edit', 'update', 'index');
+    }
+    /**
+     * Display a listing of user resource.
+     *
+     * @return Factory|View
+     */
+    public function index()
+    {
+        $users = $this->userRepository->paginate(10);
+        return view('users.index')->with(['users' => $users]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return Factory|View
+     */
+    public function create()
+    {
+        return view('users.create');
+    }
+
+    /**
+     * Store a newly created user resource in storage.
+     *
+     * @param CreateUserRequest $request
+     * @return Response
+     */
+    public function store(CreateUserRequest $request)
+    {
+        $data = $request->only('name', 'email', 'password');
+        $data['password'] = Hash::make($data['password']);
+        $data['role'] = $request->get('isAdmin');
+        User::create($data);
+        return redirect()->to('users')->withSuccess('User Account Created Successfully');
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param User $user
+     * @return Factory|View
+     */
+    public function show(User $user)
+    {
+        return view('users.show')->with(['user' => $user]);
```

Comment:
```suggestion
        return view('users.show', compact('user'));
```

Cleaner but your choice. But `compact` is best used when the name of variable matches what you want to pass to the blade file. `with(['user' => $userData])` best when you want the blade file to receive a different variable name from what you have used in the controller method.